### ISSUE TITLE
onNavigatorEvent called twice

### ISSUE THREAD

**Issue by razvan-paul** [wix/react-native-navigation#603]
Hello!

I've put the buttons in the component using the following snippet:

`this.props.navigator.setButtons({
      leftButtons: [{
          icon: iconsMap['ios-menu'],
          id: 'closeButton',
          showAsAction: 'ifRoom'
        }
      ],
      rightButtons: [{
        icon: iconsMap['ios-chatbubbles--active'],
        id: 'filterButton',
        showAsAction: 'ifRoom'
      }]
    });`

But, when the button is tapped, the following method gets called twice:

`  onNavigatorEvent(event) { 

    if (event.type == 'NavBarButtonPress') { // this is the event type for button presses
      if (event.id == 'closeButton') {
        this.props.navigator.dismissModal({
          animationType: 'slide-down'
        });
      }

      if (event.id == 'filterButton') {
        this.props.navigator.showModal({
          screen: constants.kSMFilterScreen,
          title: "Filters",
          navigatorStyle: navigationBarStyle,
          passProps: {
            filterList: this.state.filterList,
            didUpdateFilterList: this.didUpdateFilterList
          }
        })
      }
    }
  }`

My code written in onNavigatorEvent shows a modal and that component is instantiated twice.
I've tried with some anonymous functions, but nothing worked. Do you have some suggestions? What might cause this?

---
### Environment
* React Native Navigation version: 2.0.0-experimental.160
* React Native version: 0.39
* Platform(s) (iOS, Android, or both?): iOS
* Device info (Simulator/Device? OS version? Debug/Release?): iOS Simulator

**Comment by razvan-paul** [comment]
Follow Up

Apparently, this is a very strange bug. I've noticed that in some situations, the onNavigatorEvent was "called" twice, but actually it was not. There were two components constructed (I've put some logs into the constructor(props)) and that was happening when I reloaded the component using cmd+R (or double R on Windows). I don't know the reasons for this, but it happened only with react-native-navigation, so I'm just going to skip the thing when I ship my app.

Simply don't mind this situation, it's going to be fixed when you release it because users can't obviously reload the app using cmd+R haha.

**Comment by razvan-paul** [comment]
TL;DR Simply don't mind this situation, it's happening only in a debug environment.

**Comment by srameshr** [comment]
@PaulRBerg Bug never the less. How do I prevent it from firing twice on my debug env?

**Comment by PaulRBerg** [comment]
Sorry to say I didn't have any workaround back then, as I don't have one now. (as the issue is still open). It's simply the way it is and the single option is to neglect it.

**Comment by jslz** [comment]
I have seen something similar-at-first-blush today. I was getting 2 calls for every event, to onNavigatorEvent() that I had registered in componentDidMount(). The two calls had seemingly different values of 'this.props' where one of them was missing this.props.cb that was a closure. I was in debug mode. Now I can't repro after adding some console.logs() so it seems like a heisenbug.

**Comment by jslz** [comment]
btw, I am now seeing something else today where onNavigatorEvent() is only called once and the "this" pointer is not the Component but instead is the Navigator itself.
